src/Sema.zig

```diff
@@ -12259,9 +12282,17 @@ fn analyzeSwitchRuntimeBlock(
 
     const block = child_block.parent.?;
 
+    const single_prong = operand_has_opv or
+        (scalar_cases_len + multi_cases_len == 0 and !else_prong.is_inline and extra_prong == null);
+    var single_prong_payload_index: u32 = undefined;
+    assert(!(single_prong and multi_cases_len > 0));
+
     const estimated_cases_extra = (scalar_cases_len + multi_cases_len) *
         @typeInfo(Air.SwitchBr.Case).@"struct".fields.len + 2;
-    var cases_extra = try std.ArrayListUnmanaged(u32).initCapacity(gpa, estimated_cases_extra);
+    var cases_extra: std.ArrayListUnmanaged(u32) = if (single_prong)
+        .empty
```

Comment:
```suggestion
        undefined
```
makes it clear that the arraylist is completely unused